src/AutoUI/Lenses/types/table.tsx

```diff
@@ -2,13 +2,167 @@ import React from 'react';
+	border: `0.5px solid ${designTokens.color.border.accent.value}`,
+	borderRadius: '3px',
+	color: designTokens.color.text.value,
+	backgroundColor: designTokens.color.bg.accent.value,
+	position: 'relative',
+	width: 'fit-content',
+}));
+
+export interface TagLabelProps {
+	value: string;
+}
+
+export const TagLabel = ({ value }: TagLabelProps) => {
+	return (
+		<Tooltip title={value}>
+			<TagContainer>
+				<Copy copy={value} variant="absolute">
+					<Box display="flex" gap={1} mx={2} my={1}>
+						<Typography
+							variant="bodySm"
+							noWrap
+							fontWeight="bold"
+							maxWidth="100px"
+						>
+							{value}
+						</Typography>
+					</Box>
+				</Copy>
+			</TagContainer>
+		</Tooltip>
+	);
+};
+
+const tagKeyRender = (key: string) => {
+	function renderFunction(
+		tags: Array<{ tag_key: string; value: string }> | undefined,
+	) {
+		if (!tags?.length) {
+			return null;
+		}
+		const tag = tags?.find((t) => t.tag_key === key);
+		return tag ? <TagLabel value={tag.tag_key} /> : null;
+	}
+
+	return renderFunction;
+};
+
+const getResourceTags = <T extends object, P extends keyof T>(
+	item: T,
+	tagField: P,
+) => ((tagField in item ? item[tagField] : null) ?? []) as TagField[];
+
+const findTagOfTaggedResource = <T extends object>(
+	taggedResource: T,
+	tagField: keyof T,
+	tagKey: string,
+) =>
+	getResourceTags(taggedResource, tagField)?.find(
+		(tag) => tag.tag_key === tagKey,
+	);
+
+const sortData = <T extends object>(
+	data: T[],
+	columns: Array<AutoUIEntityPropertyDefinition<T>>,
+	sort: TableSortOptions | null,
+) => {
+	if (sort?.field == null) {
+		return data;
+	}
+
+	const column = columns.find((c) => c.key === sort.key);
+
+	if (!column) {
+		return data;
+	}
+
+	let collection;
+
+	if ('sortable' in column && typeof column.sortable === 'function') {
+		collection = data.slice().sort(column.sortable);
+	} else if (column.key.startsWith(TAG_COLUMN_PREFIX)) {
+		collection = data.slice().sort((a: T, b: T) => {
+			if (!column.title) {
+				return 0;
+			}
+
+			const item1tag = findTagOfTaggedResource(
+				a,
+				column.field as unknown as keyof T,
+				column.title,
+			);
+			const item2tag = findTagOfTaggedResource(
+				b,
+				column.field as unknown as keyof T,
+				column.title,
+			);
+
+			// first compare the objects
+			// so that we differentiate not having a value
+			// with not having the tag at all
+			if (!item1tag && !item2tag) {
+				return 0;
+			}
+
+			if (!item1tag) {
+				return 1;
+			}
+
+			if (!item2tag) {
+				return -1;
+			}
+
+			return (item1tag.value || '').localeCompare(item2tag.value || '');
+		});
+	} else {
+		collection = data.slice().sort((a, b) => {
+			const aValue = a[sort.field as keyof T];
+			const bValue = b[sort.field as keyof T];
+
+			if (aValue < bValue) {
+				return -1;
+			}
+			if (aValue > bValue) {
+				return 1;
+			}
+			return 0;
+		});
+	}
+
+	if (sort.direction === 'desc') {
+		collection.reverse();
+	}
```

Comment:
Wouldn't it be better for performance to swap the `-1`s and `1`s before the if-elses above if `sort.direction === 'desc'` rather than call `collection.reverse()` after?